runtime/marshal_jsonpb_test.go

```diff
@@ -270,8 +270,8 @@ func TestJSONPbEncoder(t *testing.T) {
 				if strings.ContainsAny(json, " \t\r\n") {
 					t.Errorf("strings.ContainsAny(%q, %q) = true; want false", json, " \t\r\n")
 				}
-				if strings.Contains(json, "ONE") {
-					t.Errorf(`strings.Contains(%q, "ONE") = true; want false`, json)
+				if !strings.Contains(json, "ONE") {
```

Comment:
Is this a change in behaviour?

Reply:
I think so - I assumed it was fixing a bug since the test seemed odd to me before.

In master, the behavior is different in two tests:

`TestJSONPbMarshal` tests:
- if `EnumsAsInts == false`, assert `!strings.Contains(json, "ONE")`
- if `EnumsAsInts == true`, assert `strings.Contains(json, "ONE")`

`TestJSONPbEncoder` tests:
- if `EnumsAsInts == false`, assert `!strings.Contains(json, "ONE")`
- if `EnumsAsInts == true`, assert `!strings.Contains(json, "ONE")`

The change is in the last case, but I'm not 100% sure why this changed or what the behavior ought to be. Do you have any insight into this area?

Reply:
The node_tests make this look more like a bug - there's a `TODO(yugui) Support enum by name` before `map_value: { a: 1, b: 2 }`. If you change the digits to strings, you get `unknown value \"TWO\" for enum grpc.gateway.examples.examplepb.NumericEnum`. So it seems that not supporting strings is almost intentional...

Maybe we need to change the configuration such that `jsonpb` will continue to have the old behavior.

@achew22, do you know anything about this code?

Reply:
FYI: I suspect this PR is the cause: https://github.com/golang/protobuf/pull/645

It looks like the grpc-gateway tests rely on behavior that golang/protobuf considered broken (enum values in maps).

Reply:
Great digging, I think we should obviously fix the test then.